packages/components/src/styles/components/app-side-nav/content.scss

```diff
@@ -0,0 +1,177 @@
+  color: var(--token-app-side-nav-color-foreground-primary);
+  text-decoration: none;
+  // "Link" could render as an HTML button element so this overrides the default border style in that case:
+  border-color: transparent;
+  border-radius: var(--token-app-side-nav-body-list-item-border-radius);
+
+  // :Focus
+  &:focus,
+  &.mock-focus {
+    @include hds-focus-ring-with-pseudo-element();
+  }
+
+  // :Hover
+  &:hover,
+  &.mock-hover {
+    background: var(--token-app-side-nav-color-surface-interactive-hover);
+    border-color: transparent;
+
+    .hds-app-side-nav__list-item-text,
+    .hds-app-side-nav__list-item-icon-leading,
+    .hds-app-side-nav__list-item-icon-trailing {
+      color: var(--token-app-side-nav-color-foreground-strong);
+    }
+  }
+
+  // :Active
+  &:active,
+  &.mock-active {
+    background: var(--token-color-surface-interactive-active);
+  }
+
+  // ".Active" (current page) link
+  // NOTICE: this is necessary to mitigate the flickering of the link when clicked (expecially noticeable with a badge)
+  // For details see: https://github.com/hashicorp/design-system/pull/1328
+  // &:hover:focus, // This is causing issues with :active state styles not applying, refactored to apply to "active" class
+  // notice: this ".active" extra class is used to match the corresponding `active` class assigned automatically
+  // by the `<LinkTo>` Ember component (when the link is "current), so that consumers get it for free if they want
+  // otherwise they can use the `@isActive` argument to set this visual state directly
+  // Important: This element does not doing anything when interacted with so should not change color according to state
+  &.active,
+  &.active:hover:focus {
+    background: var(--token-color-surface-strong);
+
+    // indicator bar
+    &::after {
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      left: calc(var(--token-app-side-nav-wrapper-padding-horizontal) * -1);
+      width: 4px;
+      background: var(--token-color-foreground-action);
+      border-radius: 0 2px 2px 0;
+      content: "";
+    }
+
+    .hds-app-side-nav__list-item-text,
+    .hds-app-side-nav__list-item-icon-leading,
+    .hds-app-side-nav__list-item-icon-trailing {
+      color: var(--token-color-foreground-action);
+    }
+  }
+}
+
+// special override for the "back-link" element (no visible active state, by design)
+.hds-app-side-nav__list-item-link--back-link {
+  &:active,
+  &.mock-active {
+    background: var(--token-app-side-nav-color-surface-primary);
+
+    .hds-app-side-nav__list-item-text,
+    .hds-app-side-nav__list-item-icon-leading,
+    .hds-app-side-nav__list-item-icon-trailing {
+      color: var(--token-app-side-nav-color-foreground-primary);
+    }
+  }
+}
+
+
+// LIST ITEM > INNER ELEMENTS
+
+.hds-app-side-nav__list-item-text {
+  min-width: 0;
+  max-width: 100%;
+  color: var(--token-app-side-nav-color-foreground-primary);
```

Comment:
Now we know why the question between Section Titles and List Items was looking different in code and design, because it was. So now after looking more closely I do think a change needs to be made, I'm not sure exactly how you want to do it so I won't suggest, but I think the color of the ListItems should be changed to `Foreground / Strong` just to give us a bit of extra breathing room.

Reply:
Updated, thanks